### DISCUSSION TITLE
System Prompts can not work on the first round.

### DISCUSSION THREAD

**Discussion by DirtyKnightForVi** [open-webui/open-webui#4381] [General]
# Bug Report

## Description

**Bug Summary:**
System Prompts can not work on the first round.

**Actual Behavior:**
For a specific task scenario, there might be special System Prompts. However, in the current version (at least starting from 3.10), an additional round of conversation is needed before these System Prompts can take effect.

Assuming the scenario is to generate SQL.

In the previous normal version, you could set the System Prompts in Setting --- General, then start inputting questions to generate SQL.

However, in the current version, after inputting a question, an additional round of Q&A with the LLM is required to produce the desired SQL.

## Environment

- **Open WebUI Version:** 3.11
- **Ollama (if applicable):** 0.3.3
- **Operating System:** linux
- **Browser (if applicable):** Edge

**Screenshots (if applicable):**
![微信图片_20240805180708](https://github.com/user-attachments/assets/a4f218a5-cefc-4449-9b58-7c5e5adbdf6d)
![微信图片_20240805180959](https://github.com/user-attachments/assets/b7cb9bfd-aca8-4cb8-b626-df53e611fab8)

**Comment by DirtyKnightForVi** [comment]
If I ask this question at the beginning, the large model will respond off-topic, as if the System Prompt I provided has no effect. However, when I ask the question during the second round of conversation, it can generate SQL normally.

**Comment by tjbck** [comment]
This most likely is an Ollama issue, keep us updated on your troubleshooting progress.

**Reply by DirtyKnightForVi** [reply]
i think not.
When i back to ollama 0.2.8 (the worked well version), problem still exists

**Reply by DirtyKnightForVi** [reply]
But，if i changed any paramters in `Chat Controls` (not in Settings --- General), it will works well...

**Reply by DirtyKnightForVi** [reply]
see：

![微信图片_20240805210607](https://github.com/user-attachments/assets/3c433664-58f7-436a-9de2-101ace217f3f)


![微信图片_20240805210739](https://github.com/user-attachments/assets/091a9db1-9374-4789-afa9-ac45bdb5c877)

**Reply by DirtyKnightForVi** [reply]
the 1st pic says that: I am an AI ,i can not get the new imformation balabala ···
In terms of phenomena, it is the same as generating SQL queries without providing any System Prompt.

By the way, the System Prompts in the image are exactly the same as those set in `Settings`.

**Reply by DirtyKnightForVi** [reply]
Interestingly, this phenomenon appears in most models with over 70B parameters and the recent Nemo models, even though it did not occur one or two months ago, such as with WizardLM2.

**Comment by DirtyKnightForVi** [comment]
To be honest, I don't know where the problem lies, but it is clear that OpenWebUI may have some flaws in managing the model's System Prompt.

My queries might involve sensitive, real-time financial information, so please forgive me for not being able to share those details. However, information such as schema and generation requirements is placed in the System.

Normally, I expect the model to directly respond to me with SQL, just like it did a month or two ago, instead of saying things like "I need to query real-time data" or "Sorry, I can't answer" which are unhelpful.

I have observed the following issues, especially pronounced in the WizardLM-2-8x22B model:
1. The `SYSTEM` in `ModelFile` seems to be ignored unless the model's `System Prompt` is reconfigured in `Workspace`.
2. The priority of `Chat Controls` seems to be too high, an issue I've mentioned in previous images.
3. Setting the `System Prompt` in `PlayGround`'s `Chat` mode results in the same problem as mentioned above. However, if the `System Prompt` is organized according to `TEMPLATE` in `Text Completion` mode, it can accurately generate SQL.

@tjbck

**Comment by Rudd-O** [comment]
I don't think this is an open-webui problem.  I think this is an ollama problem.  Same happens with Home Assistant -- the system prompt is ignored.

**Reply by justinh-rahb** [reply]
It wouldn't be the first, second, third, etc. time that Ollama broke system prompts.

**Reply by DirtyKnightForVi** [reply]
But ollama api is fine.
